code-challenges.test.js

```diff
@@ -10,47 +12,125 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
-
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
+const secretCodeWord1 = "Lackadaisical";
 // Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
+const secretCodeWord2 = "Gobbledygook";
 // Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
+const secretCodeWord3 = "Eccentric";
 // Expected output: "3cc3ntr1c"
 
+describe("secretMessage", () => {
+  it("takes in a string and returns a coded message", () => {
+    const secretCodeWord1 = "Lackadaisical";
+    const secretCodeWord2 = "Gobbledygook";
+    const secretCodeWord3 = "Eccentric";
 
-// b) Create the function that makes the test pass.
+    expect(secretMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l");
+    expect(secretMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k");
+    expect(secretMessage(secretCodeWord3)).toEqual("3cc3ntr1c");
+  });
+});
 
+//   ReferenceError: secretMessage is not defined, buen fracaso
+
+// b) Create the function that makes the test pass.
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
+const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"];
+const letterA = "a";
 // Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
+const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+const letterE = "e";
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe("wordReturner", () => {
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"];
+    const letterA = "a";
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+    const letterE = "e";
+
+    expect(wordReturner(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"]);
+    expect(wordReturner(arrayOfWords2)).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+  });
+});
+
+// ReferenceError: wordReturner is not defined
 
 // b) Create the function that makes the test pass.
 
+// PSEUDOCODE:
+// create function called wordReturner
+// takes in array and single letter
+// create for loop to iterate
+// if statement
+// check each index for the letter provided
+// true, return the word
```

Comment:
What does return do to a function? 

Return is the final output of a function code block. Where ever we put it it will stop to computation of the code that it exists inside of, and output whatever value it currently has.